packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -267,13 +274,13 @@ export class VirtualNode extends VirtualNodeBase {
             attributes: renderAttributes(props.cloudMapServiceInstanceAttributes),
           } : undefined,
         },
-        logging: {
+        logging: props.accessLog !== undefined ? {
           accessLog: {
             file: {
-              path: '/dev/stdout',
```

Comment:
Would defaulting the `filePath` to `/dev/stdout` be reasonable? It's the most common configuration we expect for folks using access logs.

Reply:
Let me ask you this.

Is it reasonable to leave this property unset (by that I mean `null`, as this property is optional in CloudFormation)? Is it a legitimate thing that customers might want to do?

Reply:
Path itself cannot be `null` (required in CFN: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-fileaccesslog.html). However, nulling the access log field in total is acceptable, and just means that no access logging will be configured for the resource

Reply:
So, is it reasonable for a customer to want to turn off logging by setting this (entire) property to `null`? Because if it is, we need to to take this requirement into account when designing this functionality.

I see 2 ways of allowing that:

1. Don't default this field (this way, not setting it in the CDK = passing `null` through CloudFormation).
2. If you want to go with the union-like pattern that I mentioned in [this comment](https://github.com/aws/aws-cdk/pull/10490#discussion_r493973775), you can have a "magic" `AccessLog` static factory method like `none`:

```ts
export abstract class AccessLog {
  public static none(): AccessLog { ... }

  // ...
}
```

which will set this field in the implementation of `VirtualNode` to `null`.

Reply:
For Virtual Nodes, wouldn't the equivalent to having no access logging be:

```typescript
new VirtualNode(stack, 'testVn', {
  mesh: mesh,
  virtualNodeName: 'exampleVirtualNodeName',
})
```

Then if you wanted to enable the default access logging you could do something like:

```typescript
new VirtualNode(stack, 'testVnWithDefaultAccessLogging', {
  mesh: mesh,
  virtualNodeName: 'exampleVirtualNodeName',
  accessLog: {},
})
```

or are the two equivalent?

Reply:
I think we're talking about 2 different defaults here 🙂.

Today, this:

```ts
new VirtualNode(stack, 'testVn', {
  mesh: mesh,
  virtualNodeName: 'exampleVirtualNodeName',
});
```

will result in the CDK default for logging of `{ file: { path: '/dev/stdout' } }`. I believe we're discussing in this comment whether to keep this default, or to get rid of it (see @bcelenza's original comment).

If you do this:

```ts
new VirtualNode(stack, 'testVnWithDefaultAccessLogging', {
  mesh: mesh,
  virtualNodeName: 'exampleVirtualNodeName',
  accessLog: {},
});
```

you will get the service default instead, which I believe is "no logging is enabled" (but I'm sure you know it better than I do 🙂)

Reply:
Yeah, there are actually two questions here:
1. Should we enable access logging by default?
2. Should the default path of that access log be `/dev/stdout`?

Having given this some thought, I think the answer to both is no. With `/dev/stdout`, the access logs are mixed in with the logs from the proxy process, so I think it's important folks understand the impact of that (namely they need to parse two formats from the same sink). Also, `/dev/stdout` is *nix specific and while we don't support Windows now, I can't definitively say we wouldn't support it in the future.